src/org/openj9/envInfo/MachineInfo.java

```diff
@@ -104,17 +109,79 @@ private String execCommands(String[] commands) {
 		return rt;
 	}
 
+	public static Map<String, String> requirements;
+	static {
+		requirements = new HashMap<>();
+		requirements.put("antVersion", "1.9.6");
+   	 	requirements.put("makeVersion", "4.1");
+   	 	requirements.put("perlVersion", "5.10.1");
+    	requirements.put("curlVersion", "7.20.0");
+	}
+
+	String parseInfo(String req) {
+		Pattern pattern = Pattern.compile("[0-9]+[.][0-9]+([.][0-9]+)?"); 
+		Matcher matcher = pattern.matcher(req);
+		if (matcher.find()) {
+			return matcher.group(0);
+		} else {
+			String temp = "";
+			return temp;
+		}
+	}
+
+	int[] versionStr2Arr(String ver) throws NumberFormatException {
+		String[] verSplit = ver.split("\\.");
+		int[] verArr = new int[verSplit.length];
+		for (int i=0; i < verSplit.length; i++) {
+			verArr[i] = Integer.parseInt(verSplit[i]);
+		}
+		return verArr;
+	}
+	
+	boolean validateVersion(String key, String ver, String reqVer) {
```

Comment:
Please try to use the full name as the parameter names. 
i.e., key -> versionName
ver-> actualVersionStr
reqVer -> requiredVersionStr